File: Backend/Api/Helpers/Pagination.cs
using static System.Runtime.InteropServices.JavaScript.JSType;

namespace Api.Helpers
{
    public class Pagination<T>(int pageSize, int pageIndex, int count, IReadOnlyList<T> data)
    {
        public int PageSize { get; set; } = pageSize;
        public int PageIndex { get; set; } = pageIndex;
        public int Count { get; set; } = count;
        public IReadOnlyList<T> Data { get; set; } = data;
    }
}

File: Client/src/app/shared/models/pagination.ts
export type Pagination<T> = {
  pageSize: number;
  pageIndex: number;
  count: number;
  data: T[];
};
export type WorkerParms = {
  pageIndex?: number | null;
  pageSize?: number | null;
  country?: string | null;
  rating?: number | null;
  search?: string | null;
};
export type JobParms = {
  pageIndex?: number | null;
  pageSize?: number | null;
  search?: string | null;
  budget?: number | null;
  status?: string | null;
};
export type OfferParms = {
  pageIndex?: number | null;
  pageSize?: number | null;
};
